Whisky/Views/Bottle Views/ProgramsView.swift

```diff
@@ -68,26 +80,46 @@ struct ProgramsView: View {
         .animation(.easeInOut(duration: 0.2), value: isBlocklistExpanded)
         .navigationTitle("tab.programs")
         .onAppear {
-            programs = bottle.updateInstalledPrograms()
-            blocklist = bottle.settings.blocklist
-            sortPrograms()
+            loadPrograms()
         }
         .onChange(of: resortPrograms) {
+            sortPrograms()
             reloadStartMenu.toggle()
+        }
+    }
+
+    func loadPrograms() {
+        if bottle.programs.isEmpty {
+            updatePrograms()
+            return
+        }
+
+        programs = bottle.programs
+        blocklist = bottle.settings.blocklist
+        sortPrograms()
+    }
+
+    private func updatePrograms() {
+        isLoadingPrograms = true
+
+        DispatchQueue(label: "whisky.lock.queue").async {
             programs = bottle.updateInstalledPrograms()
             blocklist = bottle.settings.blocklist
             sortPrograms()
+            isLoadingPrograms = false
         }
     }
 
-    func sortPrograms() {
-        var favourites = programs.filter { $0.pinned }
-        var nonFavourites = programs.filter { !$0.pinned }
-        favourites = favourites.sorted { $0.name < $1.name }
-        nonFavourites = nonFavourites.sorted { $0.name < $1.name }
-        programs.removeAll()
-        programs.append(contentsOf: favourites)
-        programs.append(contentsOf: nonFavourites)
+    private func sortPrograms() {
+        DispatchQueue(label: "whisky.lock.queue").async {
```

Comment:
In general, we don't use GCD in Whisky, using structured concurrency would be preferable.

Reply:
I opted for using a DispatchQueue in this specific scenario to ensure bottle loading operations get queued up in the same thread. Structured concurrency would not have this guarantee (at least as far as I know), and in general I avoided going down the route of rewriting Bottle to be an Actor.

Reply:
Actor based approach might be preferred here.